tests/foreman/ui/test_hostcollection.py

```diff
@@ -383,3 +404,223 @@ def test_negative_hosts_limit(self):
             self.assertIsNotNone(self.hostcollection.wait_until_element(
                 common_locators['alert.error_sub_form']))
+
+
+@run_in_one_thread
+class HostCollectionPackageManagementTest(UITestCase):
+    """Implements Host Collection package management related tests in UI"""
+
+    hosts_number = 2  # number of hosts per host collection
+
+    @skip_if_not_set('clients', 'fake_manifest')
+    def setUp(self):
+        """Do the following:
+
+        1. Create Org, Lifecycle Environment, Content View, Activation key
+        2. Create VMs, subscribe them to satellite-tools repo, install
+           katello-ca and katello-agent packages
+        3. Create Host collection, associate it with previously created hosts
+
+        """
+        super(HostCollectionPackageManagementTest, self).setUp()
+        self.org_ = entities.Organization().create()
+        self.env = entities.LifecycleEnvironment(
+            organization=self.org_).create()
+        self.content_view = entities.ContentView(
+            organization=self.org_).create()
+        self.activation_key = entities.ActivationKey(
+            environment=self.env,
+            organization=self.org_,
+        ).create()
+        setup_org_for_a_rh_repo({
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': self.org_.id,
+            'content-view-id': self.content_view.id,
+            'lifecycle-environment-id': self.env.id,
+            'activationkey-id': self.activation_key.id,
+        })
+        setup_org_for_a_custom_repo({
+            'url': FAKE_0_YUM_REPO,
+            'organization-id': self.org_.id,
+            'content-view-id': self.content_view.id,
+            'lifecycle-environment-id': self.env.id,
+            'activationkey-id': self.activation_key.id,
+        })
+        self.hosts = []
+        for _ in range(self.hosts_number):
+            client = VirtualMachine(distro='rhel71')
+            self.hosts.append(client)
+            client.create()
+            client.install_katello_ca()
+            result = client.register_contenthost(
+                self.org_.label, self.activation_key.name)
+            self.assertEqual(result.return_code, 0)
+            client.enable_repo(REPOS['rhst7']['id'])
+            client.install_katello_agent()
+        host_ids = [
+            entities.Host().search(query={
+                'search': 'name={0}'.format(host.hostname)})[0].id
+            for host in self.hosts
+        ]
+        self.host_collection = entities.HostCollection(
+            host=host_ids,
+            organization=self.org_,
+        ).create()
+
+    def tearDown(self):
+        """Destroy all the VMs"""
+        for client in self.hosts:
+            client.destroy()
+        super(HostCollectionPackageManagementTest, self).tearDown()
+
+    def _validate_package_installed(self, hosts, package_name,
```

Comment:
What do you think if this method becomes a hostcollection method? 

so you can use directly in any test `self.hostcollection.validate_package_installed`, so in future tests we can use it.